### PR TITLE
fix to pass sr_stridedslice_squeeze

### PR DESCRIPTION
**Pull request by ceciliapeng2011**
### Details:
 - *fix to pass sr_stridedslice_squeeze when begin_masks.size() is no larger than axis that is going to be squeezed..*
 - *corresponding unit test case.*

### Tickets:
 - *CVS-71981*

### CODE REVIEW

File: src/core/src/pass/smart_reshape/strided_slice_squeeze.cpp [hunk 1]
```diff
@@ -57,13 +56,35 @@ ngraph::pass::StridedSliceSqueeze::StridedSliceSqueeze() {
             }))
             return false;
 
+        auto const_axes_tmp = const_axes->cast_vector<int64_t>();
+
         const auto& axes = normalize_axes(squeeze->description(),
-                                          const_axes->cast_vector<int64_t>(),
-                                          squeeze->get_input_partial_shape(0).rank());
+                                          const_axes->cast_vector<int64_t>(),           // [1]
```
Comment (luo-cheng2021) [inline src/core/src/pass/smart_reshape/strided_slice_squeeze.cpp:62]: comments needs to make sense

File: src/core/src/pass/smart_reshape/strided_slice_squeeze.cpp [hunk 2]
```diff
@@ -57,13 +56,35 @@ ngraph::pass::StridedSliceSqueeze::StridedSliceSqueeze() {
...
+                                              // used along corresponding dimension.
+        new_axis_mask.resize(begin_mask.size(), 0);  // validate: All masks of StridedSlice must have the same size.
+        shrink_axis_mask.resize(begin_mask.size(), 0);
+        ellipsis_mask.resize(begin_mask.size(), 0);
+
         for (const auto& axis : axes) {
             if (begin_mask[axis]) {  // corresponding dimension of the begin input is ignored. starting from 0
                 begin_vec[axis] = 0;
                 end_vec[axis] = 1;
+                ;
```
Comment (luo-cheng2021) [inline src/core/src/pass/smart_reshape/strided_slice_squeeze.cpp:87]: should remove

File: inference-engine/tests/functional/inference_engine/transformations/sr_strided_slice_squeeze.cpp [hunk 3]
```diff
@@ -37,6 +37,64 @@ TEST(SmartReshapeTests, SS_Squeeze) {
     ASSERT_TRUE(network.getFunction()->get_parameters()[0]->get_partial_shape().compatible({2, 3}));
 }
 
+TEST(SmartReshapeTests, SS_Squeeze_partial_begin_end_mask) {
+    std::shared_ptr<ngraph::Function> f(nullptr);
```
Comment (ilyachur) [inline inference-engine/tests/functional/inference_engine/transformations/sr_strided_slice_squeeze.cpp:41]: @jane-intel Please take a look to SR changes.

File: src/core/src/pass/smart_reshape/strided_slice_squeeze.cpp [hunk 4]
```diff
@@ -57,13 +56,35 @@ ngraph::pass::StridedSliceSqueeze::StridedSliceSqueeze() {
             }))
             return false;
 
+        auto const_axes_tmp = const_axes->cast_vector<int64_t>();
```
Comment (ilyachur) [inline src/core/src/pass/smart_reshape/strided_slice_squeeze.cpp:59]: Why do we need this temporary variable?

File: src/core/src/pass/smart_reshape/strided_slice_squeeze.cpp (hunk 1 above)
Comment (ceciliapeng2011) [reply src/core/src/pass/smart_reshape/strided_slice_squeeze.cpp:62]: thanks, they are legacy of my debugging... removed.

**Review by jane-intel — APPROVED** [review]

**Pull request merged**